### PR TITLE
Revise quickstart source

### PR DESCRIPTION
**Pull request by idelgado**
- Moved all source under `com.twilio.notify.quickstart`
- Created subpackages for logical groupings
- Renamed classes to match functionality
- Simplified `BindingIntentService` logic
- Simplified Retrofit SDK starter API
- Reduced overall usage of SharedPreferences
- Pass message information to the MainActivity when binding operations complete
- Add doc comments related to Binding model and register resource
- Applied general Android style/conventions hygiene

I'm planning to use much of the same code in an example module in `video-quickstart-android`. Prior to doing so I wanted to clean things up so the developer experience matched as I ported code over to this project. I plan to use notify as an example of how to send invitations to join Video rooms.

TODO:
- [x] Do more extensive testing by uninstalling, installing, and performing different variations of binding registrations

### CODE REVIEW

**Comment by idelgado** [comment]
While much of the code is similar because of the file renames they appear as new files 😞

File: app/src/main/java/com/twilio/notify/quickstart/notifyapi/TwilioSDKStarterAPI.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.twilio.notify.quickstart.notifyapi.model.Binding;
+
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.POST;
+
+public class TwilioSDKStarterAPI {
+    private static String BASE_SERVER_URL = "https://f5ab0b54.ngrok.io";
```
Comment (jefflinwood) [inline app/src/main/java/com/twilio/notify/quickstart/notifyapi/TwilioSDKStarterAPI.java]: Can you replace this ngrok URL with YOUR_SDK_STARTER_SERVER_URL to match the README? Thanks

**Comment by jefflinwood** [comment]
Tested out these changes with the SDK Starter Kit, and it works as expected.

File: app/src/main/java/com/twilio/notify/quickstart/notifyapi/TwilioSDKStarterAPI.java (hunk 1 above)
Comment (idelgado) [inline app/src/main/java/com/twilio/notify/quickstart/notifyapi/TwilioSDKStarterAPI.java]: Derp. Forgot to remove my ngrok. Good catch.

**Comment by idelgado** [comment]
@jefflinwood finished testing and addressed feedback.

**Comment by viktormuller** [comment]
Looks good to me. Only question is if we want to do more selective sharedPreferences clear. People copy and paste this code without reading into their app and we may end up clearing preferences that they did not intend to clear.

This is in BindingIntentService.java 

sharedPreferences.edit().clear().commit();

**Comment by viktormuller** [comment]
Otherwise great stuff. Thanks @idelgado for cleaning it up. Much appreciated.

**Comment by idelgado** [comment]
@viktormuller Good point. I'll post a follow on PR to address your feedback.

**Comment by viktormuller** [comment]
Great. Thanks. And thanks @jefflinwood for merging so quickly. 👍  to everyone.

**Pull request merged**